src/spec/create_extension_spec.py

```diff
@@ -25,36 +40,189 @@ def main():
+        neurodata_type_def="NIRSDetectorsTable",
+        neurodata_type_inc="DynamicTable",
+        name="detectors",
+        doc="A table describing optical detectors of a NIRS device",
+        datasets=[
+            NWBDatasetSpec(
+                name="label",
+                doc="The label of the detector",
+                dtype="text",
+                shape=(None,),
+                neurodata_type_inc="VectorData",
+            ),
+            NWBDatasetSpec(
+                name="x",
+                doc="The x coordinate of the optical detector",
+                dtype="float",
+                shape=(None,),
+                neurodata_type_inc="VectorData",
+            ),
+            NWBDatasetSpec(
+                name="y",
+                doc="The y coordinate of the optical detector",
+                dtype="float",
+                shape=(None,),
+                neurodata_type_inc="VectorData",
+            ),
+            NWBDatasetSpec(
+                name="z",
+                doc="The z coordinate of the optical detector",
+                dtype="float",
+                shape=(None,),
+                neurodata_type_inc="VectorData",
+                quantity="?",
+            ),
+        ],
+    )
+
+    nirs_channels = NWBGroupSpec(
+        neurodata_type_def="NIRSChannelsTable",
+        neurodata_type_inc="DynamicTable",
+        name="channels",
+        doc="A table describing optical channels of a NIRS device",
+        datasets=[
+            NWBDatasetSpec(
+                name="label",
+                doc="The label of the channel",
+                dtype="text",
+                shape=(None,),
+                neurodata_type_inc="VectorData",
+            ),
+            NWBDatasetSpec(
+                name="source",
+                doc="A reference to the optical source for this channel in NIRSSourcesTable",
+                shape=(None,),
+                neurodata_type_inc="DynamicTableRegion",
+            ),
+            NWBDatasetSpec(
+                name="detector",
+                doc="A reference to the optical detector for this channel in NIRSDetectorsTable",
+                shape=(None,),
+                neurodata_type_inc="DynamicTableRegion",
+            ),
+            NWBDatasetSpec(
+                name="wavelength",
+                doc="The wavelength of light for this channel in nm",
+                dtype="float",
+                shape=(None,),
+                neurodata_type_inc="VectorData",
+                attributes=[
+                    NWBAttributeSpec(
+                        name="unit",
+                        doc="The unit of measurement for the wavelength",
+                        dtype="text",
+                        value="nm",
+                    )
+                ],
+            ),
+        ],
+    )
+
+    nirs_device = NWBGroupSpec(
+        neurodata_type_def="NIRSDevice",
+        neurodata_type_inc="Device",
+        doc="A NIRS Device",
```

Comment:
It seems like this is where we could include the NIRS contrast mechanism / mode. Would it make sense to insert it as a dataset spec? For example:
```suggestion
    nirs_device = NWBGroupSpec(
        neurodata_type_def="NIRSDevice",
        neurodata_type_inc="Device",
        doc="A NIRS Device",
        datasets=[
            NWBDatasetSpec(
                name="nirs_mode",
                doc="The mode of NIRS",
                dtype="text",
                shape=(None,),
                neurodata_type_inc="VectorData",
                quantity="continuous-wave",
            ),
         ],
         ...
```

Reply:
.. and we'd also want to include similar optional dataset entries for `time-delay`, `moment-order`, `correlation-time-delay`,`correlation-time-delay-width`...

Reply:
Browsing existing NWB specs, it looks like this kind of information could be added as text Attributes, scalar text Datasets, or in some cases as attributes on Datasets. The actual spec definition is a bit finicky and will require some experimentation to get a good fit.

As we discussed, I think it'd be easier to save this update for the next PR (but to make sure we add it before submitting to the extension catalog). Just trying to respond to this comment sent me on a 1/2hr documentation reading spree haha.

Reply:
\* adds "read documentation on text attributes, scalar text datasets, and attributes on dataset to to-do list \* (somehow I think this will take me longer than 1/2 hr haha)

Yeah, sounds good - and definitely better to have a separate PR.